File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import ceil
def solution(progresses, speeds):
    answer = []
    acc = 0
    idx = 0
    
    while idx < len(progresses):
        current = 1
        left = 100 - progresses[idx] - speeds[idx] * acc;
        acc = acc + (1 if left//speeds[idx] == 0 else ceil(left / speeds[idx]))
        next = idx + 1
        while next < len(progresses):
            if acc * speeds[next] + progresses[next] >= 100:
                idx+=1
                next+=1
                current+=1
            else:
                break
        answer.append(current)
        idx+=1
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.io.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> queue = new LinkedList<>();
        List<Integer> answer = new ArrayList<>();
        for(int i = 0; i< progresses.length; i++){
            queue.add(i);
        }
        int time = 0;
        
        while(!queue.isEmpty()){
            int count = 0;
            int idx= queue.peek();
            int timeNeed = (int) Math.ceil((double)(100 - progresses[idx] - speeds[idx] * time) / speeds[idx]);
            time += timeNeed;
            
            while(!queue.isEmpty() && progresses[queue.peek()] + time * speeds[queue.peek()] >= 100){
                count++;
                if(queue.isEmpty())break;
                queue.poll();
            }
            answer.add(count);
        }
        return answer.stream().mapToInt(x->x).toArray();
    }
}